CentralUnit/WebAPI/Controllers/RaceTrackingController.cs

```diff
@@ -89,5 +90,29 @@ public StatusCodeResult DeleteRider([FromQuery] string name)
             manager.RemoveRider(name);
             return new StatusCodeResult(200);
         }
+
+        [HttpPost]
+        [Route("[controller]/Penalty")]
+        public StatusCodeResult AddPenalty([FromBody] PenaltyEventArgs penalty)
+        {
+            manager.AddEvent(new PenaltyEventArgs(DateTime.Now, penalty.RiderName, penalty.StaffName, penalty.Reason, penalty.Seconds));
+            return new StatusCodeResult(200);
+        }
+
+        [HttpPost]
+        [Route("[controller]/DNF")]
+        public StatusCodeResult AddDNF([FromBody] ManualDNFEventArgs dnf)
+        {
+            manager.AddEvent(new ManualDNFEventArgs(DateTime.Now, dnf.RiderName, dnf.StaffName));
+            return new StatusCodeResult(200);
+        }
+
+        [HttpPost]
+        [Route("[controller]/DSQ")]
+        public StatusCodeResult AddDSQ([FromBody] DSQEventArgs dsq)
+        {
+            manager.AddEvent(new DSQEventArgs(DateTime.Now, dsq.RiderName, dsq.StaffName, dsq.Reason));
```

Comment:
Why use the rider name instead of a unique ID or beacon ID?
Matching things on human-generated strings always scares me a bit

Reply:
I agree that using the names is less than ideal. But i dont think matching a beacon is a good idea, since people might want to switch beacons between laps (a fall might crush a beacon for example) and matching on name would still make that work.

I would like to hold off on making a unique id here until we decide how the website based tracking is going to work. Ideally the ids across the systems would be compatible.